addons/account/models/account_move.py

```diff
@@ -508,9 +508,13 @@ def _onchange_journal(self):
             if new_currency != self.currency_id:
                 self.currency_id = new_currency
                 self._onchange_currency()
-        if self.state == 'draft' and self._get_last_sequence() and self.name and self.name != '/':
+        if self._must_reset_name():
```

Comment:
Not sure why you need the changes on the `_compute_name`, isn't the name always editable anyways?
I think it is awkward

Reply:
The name is not always editable :thinking:
It is only editable if:
- we validate an invoice and put it back into draft mode
- the fiduciary mode is set
If the fiduciary mode is not set, the name is computed only at validation. However, in fiduciary mode, we must compute the name as soon as the invoice is created to allow the user to simply modify the sequence number (and keep the sequence prefix). This is allowed by extending `_must_compute_name` in Enterprise.